Sources/App/Core/PackageContributorService.swift

```diff
@@ -0,0 +1,133 @@
+    private static func queryVCHistory(cacheDirPath: String, packageID: UUID?) throws -> String {
+        
+        if !Current.fileManager.fileExists(atPath: cacheDirPath) {
+            throw AppError.unexistentPackageCacheDir(packageID, cacheDirPath)
+        }
+
+        // attempt to shortlog
+        do {
+            return try Current.git.shortlog(cacheDirPath)
+        } catch {
+            throw AppError.shellCommandFailed("gitShortlog",
+                                              cacheDirPath,
+                                              "queryVCHistory failed: \(error.localizedDescription)")
+        }
+    }
+    
+    /// Parses the result of queryVCHistory into a collection of contributors
+    private static func parseVCHistory(log: String) throws -> [Contributor] {
+        var committers = [Contributor]()
+        
+        for line in log.components(separatedBy: .newlines) {
+            let log = line.split(whereSeparator: { $0 == " " || $0 == "\t"})
+            
+            if (log.count != 2) {
+                var identifier = [String]()
+                for i in 1..<(log.count - 1) {
+                    identifier.append(String(log[i]))
+                }
+                
+                let committer = Contributor(numberOfCommits: Int(log.first!) ?? 0,
+                                            email: String(log.last!),
+                                            name: identifier.joined(separator: " "))
+                committers.append(committer)
+            }
+            
+        }
+        return committers
+    }
+    
+}
+
+
+
+/// Strategy for selecting contributors based entirely on the number of commits 
+struct CommitSelector {
+    
+    static func filter(candidates: [Contributor], threshold: Float) -> [Contributor] {
+        if candidates.isEmpty {
+            return []
+        }
+        
+        let maxNumberOfCommits = candidates.max(by: { (a,b) -> Bool in
+            return a.numberOfCommits < b.numberOfCommits
+        })!.numberOfCommits
+        
+        return candidates.filter { canditate in
+            return Float(canditate.numberOfCommits) > threshold * Float(maxNumberOfCommits)
+        }
+    }
+}
+
+
+struct PackageAuthors : Encodable, Decodable, Equatable {
+    var authors : [Author]
+    var numberOfContributors : Int 
+}
+
+
+
+final class PackageContributorService {
+    
+    
+    /// Extracts the possible authors of the package according to the number of commits.
+    /// A contributor is considered an author when the number of commits is at least a 60 percent
+    /// of the maximum commits done by a contributor
+    /// - Parameters:
+    ///   - cacheDirPath: path to the cache directory where the clone of the package is stored
+    ///   - packageID: the UUID of the package
+    /// - Returns: PackageAuthors
+    static func authorExtractor(cacheDirPath: String, packageID: UUID?) throws -> PackageAuthors {
```

Comment:
Nit: I wonder if `cacheDirPath` is the best name here, maybe `gitRepositoryPath`?